src/app/storage/FabricTableImpl.ipp

```diff
@@ -100,6 +100,22 @@ struct EndpointEntryCount : public PersistentData<kPersistentBufferEntryCountByt
     }
 };
 
+// Prevent mutations from happening in TableEntryData::Serialize
+// If we just used a raw reference for TableEntryData::mEntry, C++ allows us
+// to mutate mEntry.mStorageId & mEntry.mStorageData in TableEntryData::Serialize
+// without having to do a const_cast
+template <typename T>
```

Comment:
Could you add a usage example contrasting it with a regular reference?

At a first read, since this provides both const and non-const operator-> i am trying to understand how this makes things better...

Reply:
Added more doc, does that help?

I'm totally fine with this PR not going in if it's simply causing confusion - Boris & I were concerned with the ability to accidentally mutate the payload from a const method, and this prevents that [without doing a const_cast]

The ideal solution is if C++ properly handled const - it doesn't.

The 2nd most ideal solution (splitting out mutating methods from not) is complex due to `DataAccessor` base class

This was a hackrific attempt at working around C++'s limitations 😅

Reply:
When reading this, I find this awkward code: it defines *both* const and non-const operator-> so it looks like it does  the same thing that the ref itself.

What you seem to be saying is that within a const function, we will only access `operator->() const` and that makes things const correct. If that is true, the documentation should explain that (like `const operator will be used from const methods`).

Reply:
When reading this, I find this awkward code: it defines *both* const and non-const operator-> so it looks like it does  the same thing that the ref itself.

What you seem to be saying is that within a const function, we will only access `operator->() const` and that makes things const correct. If that is true, the documentation should explain that (like `const operator will be used from const methods`).

Reply:
Feel free to resolve this as needed, was mostly drive-by review saying that this code feels complex and non-obvious - is it worth it?